x-pack/plugins/fleet/server/services/agents/upgrade.ts

```diff
@@ -170,11 +171,21 @@ async function upgradeBatch(
     return agents;
   }, []);
 
+  let sourceUri;
+  if (options.sourceUri) {
+    sourceUri = options.sourceUri;
+  } else {
+    try {
+      sourceUri = await getDefaultSourceUri(soClient);
```

Comment:
how long does it take to `getDefaultSourceUri`? this `upgradeBatch` function is called for every 10k batches, so it might be better to get it once and pass it to the function to avoid fetching multiple times.

Reply:
I think it should be fast but you're right that it could have an impact on that scale.